### PR TITLE
skip JSON filtering on non-JSON strings

### PR DESCRIPTION
**Pull request by qrux0**
Fix for https://github.com/radare/radare2/issues/7356

### CODE REVIEW

**Comment by radare** [comment]
I would move the check of the } into r2, or at least send the pr in the sdb repo. The idea is that r2 is able to indent invalid json which is useful sometimes when resding dumps or corrupted data

File: libr/cons/grep.c [hunk 1]
```diff
@@ -330,6 +330,8 @@ R_API int r_cons_grepbuf(char *buf, int len) {
 			R_FREE (cons->grep.json_path);
 		} else {
 			char *out = r_print_json_indent (buf, I (use_color), "  ");
+			if (!out)
```
Comment (radare) [inline libr/cons/grep.c]: Add braces

**Comment by qrux0** [comment]
@radare, did you mean that __~{}__  need to prettify only JSON chunks and leave non-JSON chunks unmodified?

For example this command 

```
[0x00000000]> !!/bin/echo -e 'nonjsonstuff \n line2 \n line3 {"hello" : 42, "world":3}'~{}
```
should produce such output?
```
nonjsonstuff 
 line2 
 line3
{
  "hello": 42,
  "world": 3
}
```

**Comment by radare** [comment]
in fact 'true' or '[1,2]' is also valid json.

**Comment by radare** [comment]
Yeah maybe that makes more sense. Or st least dont eat spaces and newlines which is the reason this looks bad. But actualy i dont see a good/easy approach here. Let me know if you have a better idea

**Comment by radare** [comment]
i think the correct fix for this consists in not eating any space or tab or newline if this char is not between a pair of braces. this is... before { or after the last }. this way we can indent properly and not mess with the non -json stuff. as long as json array objects can start with [] we can also consider those as valid boundary checkers for this condition. Can you update the PR with this change?

**Comment by qrux0** [comment]
Hello, @radare.
I'm sorry, but my current pull  are already acts as you described.
I.e. if current indent level lower than 0, we just print current character without any formatting.

**Comment by radare** [comment]
oh ! didnt saw the last changes in your branch, missed that.

**Comment by radare** [comment]
there are 10 broken tests because of changes in master, can you please rebase to confirm no brokenness? thanks!

**Comment by radare** [comment]
Awesome! can you submit a test to verify that this feature works as expected and its not broken in the future? thanks!

**Pull request merged**